lib/models/release.ts

```diff
@@ -389,6 +389,56 @@ const getReleaseModel = function (
 		);
 	}
 
+	/**
+	 * @summary Promotes a 'draft' release to 'final' (non-draft)
+	 * @name finalize
+	 * @public
+	 * @function
+	 * @memberof balena.models.release
+	 *
+	 * @param {String|Number} commitOrId - release commit (string) or id (number)
+	 * @fulfil {void}
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.release.finalize(123).then(function() {
+	 * 	console.log('finalized!');
+	 * });
+	 *
+	 * @example
+	 * balena.models.release.finalize('7cf02a6').then(function() {
+	 * 	console.log('finalized!');
+	 * });
+	 *
+	 */
+	async function finalize(commitOrId: string | number): Promise<void> {
+		if (commitOrId == null) {
+			throw new errors.BalenaReleaseNotFound(commitOrId);
+		}
```

Comment:
Let's use `const { id } = await models.release.get(commitOrId, { $select: 'id' });` and use the returned `id` so that we also automatically throw in case more than one releases are returned.

Reply:
I had thought about doing something like this, but it's significantly less performant.  Probably makes no difference, and as you mention, safer.

Reply:
I also had complains about this pattern, but:
* we re-use the same "what to query on" logic
* it also throws 404 if the item isn't found (pine doesn't do that for PATCHes, but it should)